components/vocabularies/api/src/main/java/org/phenotips/vocabulary/internal/solr/AbstractOWLSolrVocabulary.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+package org.phenotips.vocabulary.internal.solr;
+
+import org.phenotips.vocabulary.VocabularyTerm;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.client.solrj.response.QueryResponse;
+import org.apache.solr.common.SolrDocument;
+import org.apache.solr.common.SolrDocumentList;
+import org.apache.solr.common.SolrException;
+import org.apache.solr.common.SolrInputDocument;
+
+import com.hp.hpl.jena.ontology.OntClass;
+import com.hp.hpl.jena.ontology.OntModel;
+import com.hp.hpl.jena.ontology.OntModelSpec;
+import com.hp.hpl.jena.rdf.model.ModelFactory;
+
+/**
+ * Ontologies processed from OWL files share much of the processing code.
+ *
+ * @version $Id$
+ * @since 1.2M4 (under different names since 1.1)
+ */
+public abstract class AbstractOWLSolrVocabulary extends AbstractSolrVocabulary
+{
+    protected static final String VERSION_FIELD_NAME = "version";
+
+
+    @Override
+    public VocabularyTerm getTerm(String id) {
+        VocabularyTerm result = super.getTerm(id);
+        if (result == null) {
+            Map<String, String> queryParameters = new HashMap<>();
+            queryParameters.put("id", id);
```

Comment:
So when would the ```super.getTerm(id)``` ever return results? Would ```ID:myid``` format still be present in OWL ontologies?